course3/homework-uniqueptr/test.cc

```diff
@@ -0,0 +1,20 @@
+#include "unique_ptr.h"
+#include <iostream>
+#include <memory>
+#include <cstring>
+
+//template<typename template<typename T> class PtrType>
+template <typename PtrType>
+void test(){
+	PtrType a(new char[5]);
+	std::strcpy(a.get(),"abcd");
+	std::cout << "value of a=" << a.get() <<std::endl;	
+}
+
+int main(){
+	std::cout<<"test std::unique_ptr"<<std::endl;
+	test<std::unique_ptr<char[]>>();
+	std::cout<<std::endl;
+	std::cout<<"test MyUniquePtr"<<std::endl;
+	test<MyUniquePtr<char[]>>();
```

Comment:
1. Don't count on compiler ignoring the difference between ```char``` and ```char[]```. They are _always_ different types, and, sadly, in different contexts the [] means different thing.
std::unique_ptr provides _specialization_ for [] types, which stores arrays instead of single objects. If you don't know how to create such a specialization, please, look for inspiration in the code of std::unique_ptr.

2. Template templates are nightmare fuel. The solution you used is a valid one.